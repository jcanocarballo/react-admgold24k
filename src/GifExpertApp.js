import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

function GifExpertApp({defaultCategories = []}) {  
  const [categories, setCategories] = useState( defaultCategories );

  // const handleAdd = () => {
  //   setCategories([...categories, 'X Men'])
  // }

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories}></AddCategory>
      <hr></hr>
      
      <ol>
        {categories.map( category => (
          <GifGrid key={category} category={category}></GifGrid>
        ))}          
      </ol>
    </>
  )
}

export default GifExpertApp

