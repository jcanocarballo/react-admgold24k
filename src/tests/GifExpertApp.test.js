import React from 'react';
import {shallow} from 'enzyme'
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en componente GifExpertApp', () => {
  test('debe mostrarse correctamente', () => {
    const wrapper = shallow(<GifExpertApp></GifExpertApp>)
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories}></GifExpertApp>);
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
  
  
})