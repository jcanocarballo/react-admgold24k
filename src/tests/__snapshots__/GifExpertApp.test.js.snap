// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifExpertApp debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
